platform/broadcom/sonic-platform-modules-juniper/qfx5200/modules/adt7470.c

```diff
@@ -0,0 +1,1375 @@
+/*
+ * A hwmon driver for the Analog Devices ADT7470
+ * Copyright (C) 2007 IBM
+ *
+ * Author: Darrick J. Wong <darrick.wong@oracle.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
```

Comment:
i can see linux has this driver. why not use the one from linux? 

https://github.com/torvalds/linux/blob/master/drivers/hwmon/adt7470.c

put into sonic-linux-kernel

Reply:
@lguohan There are some platform specifc changes needed in QFX500 platform to make the adt7470 driver work. These changes will not be applicable for the other platforms. So we had to make a private copy of the driver adt7470.

Reply:
@lguohan The default kernel driver adt7470 will not be able to adjust the fan speed in QFX5200, so we had to adjust a few parameters in the driver to make it functional in QFX5200.

Reply:
what is the nature of these private changes? why you won't be able to make it as a generic kernel patch?

Reply:
@lguohan We had a patch for adjusting the PWM frequency using the 0x40 register. Initially @ashishb-juniper  added it for QFX5200 and now I see that it's already in the mainline: https://patchwork.kernel.org/patch/9266607/

I've removed the private driver. Also tested the switch by loading the default adt7470 driver. Updating the pull request.